src/video/cocoa/SDL_cocoametalview.m

```diff
@@ -100,7 +102,9 @@ - (instancetype)initWithFrame:(NSRect)frame
 
 - (void)dealloc
 {
-    SDL_RemoveWindowEventWatch(SDL_WINDOW_EVENT_WATCH_EARLY, SDL_MetalViewEventWatch, (__bridge void *)(self));
+    if (SDL_GetHintBoolean(SDL_HINT_VIDEO_METAL_AUTO_RESIZE_DRAWABLE, true)) {
```

Comment:
We shouldn't check the hint here, we should either remember that we added the watcher or just call SDL_RemoveWindowEventWatch() unconditionally, in case someone changes the hint at runtime.